pkg/apis/core/validation/validation.go

```diff
@@ -1835,8 +1835,8 @@ func ValidatePersistentVolumeClaimSpec(spec *core.PersistentVolumeClaimSpec, fld
 			allErrs = append(allErrs, field.Required(fldPath.Child("dataSource", "name"), ""))
 		} else if !supportedDataSourceKinds.Has(string(spec.DataSource.Kind)) {
 			allErrs = append(allErrs, field.NotSupported(fldPath.Child("dataSource"), spec.DataSource.Kind, supportedDataSourceKinds.List()))
-		} else if !supportedDataSourceAPIGroups.Has(string(spec.DataSource.APIGroup)) {
-			allErrs = append(allErrs, field.NotSupported(fldPath.Child("dataSource"), spec.DataSource.APIGroup, supportedDataSourceAPIGroups.List()))
+		} else if spec.DataSource.APIGroup != nil && !supportedDataSourceAPIGroups.Has(string(*spec.DataSource.APIGroup)) {
```

Comment:
this doesn't look correct… shouldn't we be validating the combination of apiGroup and kind? you can make nonsense combinations of the two if you only validate each is in the allowed set individually

Reply:
Good point.  Will change this.

Reply:
Fixed.  @liggitt Can you please take a look?  Thanks! https://github.com/kubernetes/kubernetes/pull/68419/commits/c5e0efe3a6b31b585655ebe61e79e2c888192c2e